pt/lessons/libraries/mox.md

```diff
@@ -0,0 +1,222 @@
+---
+version: 1.0.0
+title: Mox
+---
+
+Mox é uma biblioteca feita para projetar mocks simultâneos em Elixir.
+
+{% include toc.html %}
+
+## Escrever Código Testável
+
+Os testes e os mocks que os facilitam não são, normalmente, o destaque de qualquer língua, e, por isso, não é surpreendente que exista menos literatura sobre eles.
+No entanto, pode-se _absolutamente_ usar mocks em Elixir!
+A metodologia exacta pode ser um pouco diferente da que está familiarizado noutras línguas, mas o objetivo final é o mesmo: os mocks podem simular o output de funções internas e, então, permitem-lhe assumir todas as possíveis execuções do seu código.
+
+Antes de mostrarmos use cases complexos, vamos falar de algumas técnicas que lhe poderão ajudar a tornar o seu código mais testável.
+Uma tática simples é passar um módulo para uma funções em vez de fazer hard-code do módulo dentro da função.
```

Comment:
```suggestion
Uma tática simples é passar um módulo para uma função em vez de ter o módulo hard-coded dentro da função.
```
I'm not sure in the approach here. Would be nice if we had a translation for "hard-coding".

Reply:
I tried to google for portuguese articles with `hard-coded` in it to see possible translations but really couldn't find anything suitable

Reply:
Maybe `Uma tática simples é passar um módulo para uma função em vez de ter o módulo diretamente dentro da função.` but I don't think it properly passes the intended message

Reply:
I remember a long time ago, in Brazil, a friend of mine used to say `engessado` that comes from "teto de gesso" lol, but I don't think it's applicable for this context. I'd probably consider leaving `hard-coded` for now and it should be ok.

Reply:
Sounds good to me, I completely agree with that!